backend/app/services/implementations/user_service.py

```diff
@@ -0,0 +1,88 @@
+import logging
+
+import firebase_admin.auth
+from fastapi import HTTPException
+from sqlalchemy.orm import Session
+
+from app.models import User
+from app.schemas.user import UserCreate, UserInDB, UserRole
+from app.services.interfaces.user_service import IUserService
+
+
+class UserService(IUserService):
+    def __init__(self, db: Session):
+        self.db = db
+        self.logger = logging.getLogger(__name__)
+
+    async def create_user(self, user: UserCreate) -> UserInDB:
+        firebase_user = None
+        try:
+            # Create user in Firebase
+            firebase_user = firebase_admin.auth.create_user(
```

Comment:
I think we should eventually create another function or even another service to handle firebase interaction. The firebase error handling makes it hard to read this function.